cowbird/database/mongodb.py

```diff
@@ -0,0 +1,130 @@
+# MongoDB
+# http://docs.pylonsproject.org/projects/pyramid-cookbook/en/latest/database/mongodb.html
+import warnings
+from typing import TYPE_CHECKING
+from urllib.parse import urlparse
+
+import pymongo
+
+from cowbird.database.base import DatabaseInterface
+from cowbird.database.stores import MonitoringStore, StoreInterface
+from cowbird.utils import get_settings
+
+# pylint: disable=C0103,invalid-name
+MongoDB = None  # type: Optional[Database]
+MongodbStores = frozenset([
+    MonitoringStore,
+])
+
+if TYPE_CHECKING:
+    from typing import Any, Optional, Type, Union
+
+    from pymongo.database import Database
+
+    from cowbird.database.base import StoreSelector
+    from cowbird.typedefs import JSON, AnySettingsContainer
+
+    # pylint: disable=E0601,used-before-assignment
+    AnyMongodbStore = Union[MongodbStores]
+    AnyMongodbStoreType = Union[
+        StoreSelector,
+        AnyMongodbStore,
+        Type[MonitoringStore],
+    ]
+
+
+class MongoDatabase(DatabaseInterface):
+    _database = None
+    _settings = None
+    _stores = None
+    type = "mongodb"
```

Comment:
Since I expect we'll have quite a few types eventually in this module, I think formalizing our types could be useful, both to prevent typo errors or having to do string validations/comparisons, as well as having an easy reference for the different types used by the different stores/databases. 

Maybe through something like Enum, or through constants?

Reply:
In fact `MongoDatabase.type` is the constant and the only place the literal string will ever be used. Same thing for the `MonitoringStore.type`. The list of stores is available here : https://github.com/Ouranosinc/cowbird/blob/catalog_monitoring_impl/cowbird/database/mongodb.py#L15 and by looping over `MongodbStores` I can get the `type` for each of them. Given I would use another type of database I would create another frozenset of `DatabaseInterface`